automationtestapp/src/main/java/com/microsoft/aad/automation/testapp/MainActivity.java

```diff
@@ -50,17 +53,28 @@
     public static final int ACQUIRE_TOKEN_SILENT = 1002;
     public static final int INVALIDATE_ACCESS_TOKEN = 1003;
     public static final int INVALIDATE_REFRESH_TOKEN = 1004;
+    public static final int INVALIDATE_FAMILY_REFRESH_TOKEN = 1006;
     public static final int READ_CACHE = 1005;
-
-    private Context mContext;
+    
+    private StringBuffer mADALLogs;
+    private Context mContext;    
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-
         mContext = getApplicationContext();
+        
+        mADALLogs = new StringBuffer();
+        Logger.getInstance().setExternalLogger(new Logger.ILogger() {
```

Comment:
I think you can set the external logger in the Application class, that way we won't need to have it in both MainActivity and SigninActivity.